pkg/agent/nodeportlocal/k8s/npl_controller.go

```diff
@@ -378,18 +378,13 @@ func (c *NPLController) processNextWorkItem() bool {
 	return true
 }
 
-func (c *NPLController) deleteAllPortRulesIfAny(podKey string) error {
-	klog.InfoS("Deleting all NodePortLocal rules for Pod", "pod", podKey)
-	return c.portTable.DeleteRulesForPod(podKey)
-}
-
 // handleRemovePod removes rules from port table and
 // rules programmed in the system based on implementation type (e.g. IPTABLES).
 // This also removes Pod annotation from Pods that are not selected by Service annotation.
 func (c *NPLController) handleRemovePod(key string) error {
 	klog.V(2).Infof("Got delete event for Pod: %s", key)
 
-	if err := c.deleteAllPortRulesIfAny(key); err != nil {
+	if err := c.cleanupPodRules(key, nil); err != nil {
```

Comment:
I suppose you should pass sets.New[string]() instead of nil? otherwise, the call in `podPortsToKeep.Has(util.BuildPortProto(fmt.Sprint(data.PodPort), proto.Protocol))` will panic.

Reply:
My bad memory about the sets.New[string] type.